### PR TITLE
8227257: javax/swing/JFileChooser/4847375/bug4847375.java fails with AssertionError

### PR DESCRIPTION
**Pull request by mrserb**
This assertion happens when we try to build the path linked by the shortcut file when part of that path is not accessible. For example, if the ".lnk" file is pointed to the "windows-update. settings" inside the "Internet.settings". 

After the change, we will skip the target of the link if any part of the path to the target is not a folder and is not traversable in JFileChooser.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8227257](https://bugs.openjdk.org/browse/JDK-8227257): javax/swing/JFileChooser/4847375/bug4847375.java fails with AssertionError

### Reviewers
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11789/head:pull/11789` \
`$ git checkout pull/11789`

Update a local copy of the PR: \
`$ git checkout pull/11789` \
`$ git pull https://git.openjdk.org/jdk pull/11789/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11789`

View PR using the GUI difftool: \
`$ git pr show -t 11789`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11789.diff">https://git.openjdk.org/jdk/pull/11789.diff</a>

</details>

### CODE REVIEW

**Comment by aivanov-jdk** [comment]
So, to reproduce this issue, one has to put a shortcut (`.lnk` file) which points to a virtual folder which has no file system path, any Control Panel applets or Control Panel itself could trigger this.

Is my understanding correct?

**Comment by mrserb** [comment]
> So, to reproduce this issue, one has to put a shortcut (`.lnk` file) which points to a virtual folder which has no file system path, any Control Panel applets or Control Panel itself could trigger this.
> 
> Is my understanding correct?

The user should actually open that 'place in Control Panel" first so the link will have appeared in the "recent files" folder, the exception occurs when the JFC tries to build a "files tree". This is how I get the lnk file for the test.

**Comment by aivanov-jdk** [comment]
> What do people mean by Control Panel ? Do you mean Windows (File) Explorer ? If you do mean Control Panel I think I need a picture or two to show me how that affects FileChooser. Is that lnk file format something that is "stable" across Windows releases ?

I mean the classical Control Panel that's launched in Windows Explorer as opposed to the modern Settings app.

A shortcut, `.lnk`, can point to any object in the Shell namespace (a COM object) not only to objects on file system.

From [the linked email](https://mail.openjdk.org/pipermail/awt-dev/2019-November/015494.html), I infer there's `The Internet` object (on a desktop? ) which is marked with `ATTRIB_BROWSABLE` but not `ATTRIB_FOLDER`. That is the Windows (File) Explorer can navigate to this object, yet JFileChooser can't because this object is not a file system object.

If I understand @mrserb's scenario where he reproduced the problem, one of the folders (Recent) contained a shortcut which has the same or similar properties: the File Explorer can navigate to the object to which the shortcut points, yet JFileChooser can't because it shows and navigates file system objects object.

A `.lnk` object is stable across Windows releases, basically it's a serialised form of `IShellLink` interface. Yet the shell would fail to resolve the `.lnk` if the shell extension it points to is not installed.

File: test/jdk/javax/swing/JFileChooser/4847375/bug4847375.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 /*
  * @test
- * @bug 4847375 8171363
+ * @bug 4847375 8171363 8227257
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/JFileChooser/4847375/bug4847375.java:26]: Could you also add an option to run this test with assertions when using jtreg directly like you did for the new test?

**Review by aivanov-jdk — APPROVED** [review]
Indeed, this resolves the problem. And I reproduced the original issue with `AssertionError` by running `bug4847375.java` with enabled assertions.

You're right, the Recent folder contains a handful of link files which point to the new Setting app which use `ms-settings:` protocol, File Explorer displays the *target location* as `"The Internet"`.

**Comment by aivanov-jdk** [comment]
By the way, `JFileChooser` can navigate the `.lnk` files which point to the classic Control Panel. I created a shortcut to *Devices and Printers* folder on the desktop. Double-clicking it, opens the folder but shows no items; the combo box at the top shows the structure: *Desktop* → *Control Panel* → *Hardware and Sound* → *Devices and Printers* followed by *This PC* and other common items.

File: test/jdk/javax/swing/JFileChooser/4847375/bug4847375.java (hunk 1 above)
Comment (aivanov-jdk) [reply test/jdk/javax/swing/JFileChooser/4847375/bug4847375.java:26]: Could you update the copyright year in other modified files too?

**Review by aivanov-jdk — APPROVED** [review]

**Review by TejeshR13 — APPROVED** [review]
Looks Good to me.

**Pull request closed without merging**